Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -65,57 +65,181 @@ $LogFile = "ExchAvExclusions.log"
+        }
+    }
+    $BaseFolders.Add((Join-Path $ExchangePath '\ClientAccess\OAB').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\FIP-FS').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\GroupMetrics').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\Logging').tolower())
+    $BaseFolders.Add((Join-Path $ExchangePath '\Mailbox\MDBTEMP').tolower())
+
+    $mbxS = Get-MailboxServer -Identity $($env:COMPUTERNAME) | Select-Object CalendarRepairLogPath, LogPathForManagedFolders, `
+        DataPath, MigrationLogFilePath, TransportSyncLogFilePath, TransportSyncMailboxHealthLogFilePath
+    $mbxS.psobject.Properties.Value.PathName | ForEach-Object {
+        if ( $_ ) {
+            if ( Test-Path $_ -PathType Container ) {
+                $BaseFolders.Add($_.tolower())
+            }
+        }
+    }
+
     # Add all database folder paths
     foreach ($Entry in (Get-MailboxDatabase -Server $Env:COMPUTERNAME)) {
         $BaseFolders.Add((Split-Path $Entry.EdbFilePath -Parent).tolower())
-        $BaseFolders.Add(($Entry.LogFolderPath.pathname.tolower()))
+        $mbdblogs = $Entry | Select-Object TemporaryDataFolderPath, LogFolderPath
+
+        $mbdblogs.psobject.Properties.Value.PathName | ForEach-Object {
+            if ( $_ ) {
+                if ( Test-Path $_ -PathType Container ) {
+                    $BaseFolders.Add($_.tolower())
+                }
+            }
+        }
+    }
+
+    $mtsLogs = Get-MailboxTransportService $($env:COMPUTERNAME) | Select-Object ConnectivityLogPath, `
+        ReceiveProtocolLogPath, SendProtocolLogPath, MailboxSubmissionAgentLogPath, MailboxDeliveryAgentLogPath, `
+        DnsLogPath, RoutingTableLogPath, SyncDeliveryLogPath, MailboxDeliveryHttpDeliveryLogPath, `
+        MailboxDeliveryThrottlingLogPath, AgentGrayExceptionLogPath, PipelineTracingPath
+    $mtsLogs.psobject.Properties.Value.PathName | ForEach-Object {
+        if ( $_ ) {
+            if ( Test-Path $_ -PathType Container ) {
+                $BaseFolders.Add($_.tolower())
+            }
+        }
+    }
+
+    #'$env:SystemRoot\Temp\OICE_<GUID>'
+    $possibleOICEFolders = Get-ChildItem $env:SystemRoot\temp -Directory -Filter OICE_*.0
+    $possibleOICEFolders | ForEach-Object {
+        if ( $_.Name.Length -gt 41) {
+            $possibleGUID = $_.Name.Substring(5, 36)
+            $result = [System.Guid]::Empty
+            if ( [System.Guid]::TryParse($possibleGUID, [System.Management.Automation.PSReference]$result) ) {
+                $BaseFolders.Add($_.FullName.tolower())
+            }
+        }
+    }
+}
+
+if ((Get-ExchangeServer $env:COMPUTERNAME).IsUnifiedMessagingServer) {
+    $BaseFolders.Add((Join-Path $ExchangePath '\UnifiedMessaging\Grammars'))
+    $BaseFolders.Add((Join-Path $ExchangePath '\UnifiedMessaging\Prompts'))
+    $BaseFolders.Add((Join-Path $ExchangePath '\UnifiedMessaging\Temp'))
+    $BaseFolders.Add((Join-Path $ExchangePath '\UnifiedMessaging\Voicemail'))
+}
```

Comment:
Don't believe these exist in Exchange 2019.

Reply:
it does not and it is not a IsUnifiedMessagingServer it is not going to go into, but I think it exists as a property

Reply:
I actually missed the `IsUnifiedMessagingServer` piece